pandas/tests/frame/test_constructors.py

```diff
@@ -2398,6 +2398,12 @@ def check_views():
             # assert b[0] == 0
             assert df.iloc[0, 2] == 0
 
+    def test_from_series_with_name_with_columns(self):
+        # GH 7893
+        result = DataFrame(Series(1, name="foo"), columns=["bar"])
+        expected = DataFrame(columns=["bar"])
+        tm.assert_frame_equal(result, expected)
```

Comment:
this strikes me as weird.  if the series's name was "bar", then result would have length 1 instead of 0.

Reply:
Yeah I'm not sure; just seems it was the preferred behavior from https://github.com/pandas-dev/pandas/issues/7893#issuecomment-50826071

Reply:
cc @jreback as you had mentioned that this behavior was expected

Reply:
yeah we should prob deprecate this but its the current behavior